crates/sncast/src/helpers/fee.rs

```diff
@@ -92,44 +95,44 @@ impl FeeArgs {
                         bail!("--max-fee should be greater than or equal to --max-gas-unit-price")
                     }
                     (None, _, _) => FeeSettings::Strk {
-                        max_gas: self.max_gas.map(TryIntoConv::try_into_).transpose()?,
+                        max_gas: self
+                            .max_gas
+                            .and_then(|val| NonZeroU64::try_from_(Felt::from(val)).ok()),
                         max_gas_unit_price: self
                             .max_gas_unit_price
-                            .map(TryIntoConv::try_into_)
-                            .transpose()?,
-                    },
-                    (Some(max_fee), None, Some(max_gas_unit_price)) => FeeSettings::Strk {
-                        max_gas: Some(
-                            max_fee
-                                .floor_div(&NonZeroFelt::from_felt_unchecked(max_gas_unit_price))
-                                .try_into_()?,
-                        ),
-                        max_gas_unit_price: Some(max_gas_unit_price.try_into_()?),
-                    },
-                    (Some(max_fee), Some(max_gas), None) => FeeSettings::Strk {
-                        max_gas: Some(max_gas.try_into_()?),
-                        max_gas_unit_price: Some(
-                            max_fee
-                                .floor_div(&NonZeroFelt::from_felt_unchecked(max_gas))
-                                .try_into_()?,
-                        ),
+                            .and_then(|val| NonZeroU128::try_from_(Felt::from(val)).ok()),
                     },
+                    (Some(max_fee), None, Some(max_gas_unit_price)) => {
+                        let max_gas = NonZeroFelt::try_from(Felt::from(max_fee).floor_div(&max_gas_unit_price)).context("Calculated max gas from provided --max-fee and --max-gas-unit-price is zero. Please increase --max-fee to obtain a positive gas amount")?;
+                        FeeSettings::Strk {
+                            max_gas: NonZeroU64::try_from_(Felt::from(max_gas)).ok(),
```

Comment:
This `.ok()` that you are using is something we really don’t want to do.  
If the conversion fails, we should fail the command and not proceed with the `None` variant.

Please change all occurrences